src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1710,6 +1710,48 @@ public async Task<bool> IsAsyncMethod(SessionId sessionId, int methodId, Cancell
             return retDebuggerCmdReader.ReadByte() == 1 ; //token
         }
 
+        public async Task<JArray> GetHoistedLocalVariables(SessionId sessionId, List<int> objectAlreadyRead, JArray asyncLocals, CancellationToken token)
+        {
+            JArray asyncLocalsFull = new JArray();
+            foreach (var asyncLocal in asyncLocals)
+            {
+                if (asyncLocal["name"].Value<string>().EndsWith("__this"))
+                {
+                    asyncLocal["name"] = "this";
+                    asyncLocalsFull.Add(asyncLocal);
+                }
+                else if (asyncLocal["name"].Value<string>().StartsWith("<>8__") ||
+                        asyncLocal["name"].Value<string>().StartsWith("<>f__ref") ||
+                        asyncLocal["name"].Value<string>().StartsWith("$locvar") ||
+                        asyncLocal["name"].Value<string>().StartsWith("CS$<>")) //same code that has on debugger-libs
```

Comment:
Where is the similar code in debugger-libs?

Reply:
https://github.com/mono/debugger-libs/blob/08268b730a73cca0e3b3ac9c703f0b773a3be25b/Mono.Debugging.Soft/SoftDebuggerAdaptor.cs#L865

Reply:
thanks! That code has nice method names for the conditions too. *nit*: maybe move the conditions into similarly named methods, so it's kinda documented what they are for?

Reply:
> thanks! That code has nice method names for the conditions too. _nit_: maybe move the conditions into similarly named methods, so it's kinda documented what they are for?

Adding to comments would be fine too.

Reply:
Done.